apps/04TowersOfHanoi.js

```diff
@@ -13,42 +13,108 @@ var stacks = {
   c: []
 };
 
+var moveCount = 1;
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Prints the starting board
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-}
+// Accepts two inputs: first is the stack where the piece starts and the second is the stack where the piece will end up
+function movePiece(startStack,endStack) {  
 
-function isLegal() {
-  // Your code here
+// Assigns the variable "poppedPiece" to the value of the piece that has been popped from Stack A  
+  var poppedPiece = stacks[startStack].pop();
 
+// Adds the popped piece to the top of the end stack
+  stacks[endStack].push(poppedPiece);
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Checks to see if the move is legal or illegal
+function isLegal(startStack, endStack) {
+  var start = stacks[startStack].length;
+  var end = stacks[endStack].length;
+// If the end stack's length is zero or the start stack is less than the end stack than the move is legal 
+  if (stacks[endStack].length === 0 || stacks[startStack][start - 1] < stacks[endStack][end - 1]) {
+    return true;
+  }
+//If the move is illegal, the previous move will be reversed by popping the endStack back on to the startStack  
+  else {
+    var wrongMove = stacks[endStack].pop();
```

Comment:
If you want to "pop back a bad move", you need to call `movePiece()` before `isLegal()`.

You can avoid all this by only excecuting `movePiece()` IF `isLegal()` returns true.